infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -129,6 +134,19 @@ protected OpenShiftEnvironment doCreate(
       }
     }
 
+    if (deployments.size() + pods.size() > 1) {
```

Comment:
Only semi-related to this PR but this duplication is getting quite bad..

Reply:
I know, but I have no idea how to refactor it easily to be able to handle it in this PR